robolectric/src/test/java/org/robolectric/shadows/ShadowActivityTest.java

```diff
@@ -259,18 +263,21 @@ public void startActivityForResultAndReceiveResult_shouldSendResponsesBackToActi
 
   @Test
   public void startActivityForResultAndReceiveResult_whenNoIntentMatches_shouldThrowException() {
-    ThrowOnResultActivity activity = Robolectric.buildActivity(ThrowOnResultActivity.class).get();
-    activity.startActivityForResult(new Intent().setType("audio/*"), 123);
-    activity.startActivityForResult(new Intent().setType("image/*"), 456);
-
-    Intent requestIntent = new Intent().setType("video/*");
-    try {
-      shadowOf(activity)
-          .receiveResult(
-              requestIntent, Activity.RESULT_OK, new Intent().setData(Uri.parse("content:foo")));
-      fail();
-    } catch (Exception e) {
-      assertThat(e.getMessage()).startsWith("No intent matches " + requestIntent);
+    try (ActivityController<ThrowOnResultActivity> controller =
+        Robolectric.buildActivity(ThrowOnResultActivity.class)) {
+      ThrowOnResultActivity activity = controller.get();
+      activity.startActivityForResult(new Intent().setType("audio/*"), 123);
+      activity.startActivityForResult(new Intent().setType("image/*"), 456);
+
+      Intent requestIntent = new Intent().setType("video/*");
+      try {
+        shadowOf(activity)
+            .receiveResult(
+                requestIntent, Activity.RESULT_OK, new Intent().setData(Uri.parse("content:foo")));
+        fail();
+      } catch (Exception e) {
```

Comment:
nit: we can move this `catch (Exception e)` to outer `try` statement, and remove this internal `try-catch`. The `try-with-resource` statement will auto call resource's `close` method at finally, see https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html.